client-tools/Shell/TelemetricsHandler.cpp

```diff
@@ -300,6 +300,7 @@ TelemetricsHandler::buildHttpClient(std::string& url) const {
     std::unique_ptr<httpclient::GeneralClientConnection> connection(
         httpclient::GeneralClientConnection::factory(
             cf.getCommFeaturePhase(), newEndpoint, 30, 60, 3, sslProtocol));
+    connection->setSocketNonBlocking(true);
```

Comment:
This is not safe, because the returned `connection` object can be a nullptr.
Please move the line into the following `if` statement.